### PR TITLE
Make recovery sync handle retry response

### PR DESCRIPTION
**Pull request by amstone326**

### CODE REVIEW

File: app/unit-tests/src/org/commcare/android/tests/DataPullTaskTest.java [hunk 1]
```diff
@@ -80,6 +80,13 @@ public void dataPullRecoverFailTest() {
     }
 
     @Test
+    public void dataPullRecoverWithRetryTest() {
+        installLoginAndUseLocalKeys();
+        runDataPull(new Integer[]{412, 202, 200}, new String[]{GOOD_RESTORE, RETRY_RESPONSE, GOOD_RESTORE});
```
Comment (ctsims) [inline app/unit-tests/src/org/commcare/android/tests/DataPullTaskTest.java]: Oh man this rules

File: app/src/org/commcare/tasks/DataPullTask.java [hunk 2]
```diff
@@ -465,26 +467,39 @@ private static void recordSuccessfulSyncTime(String username) {
 
     //TODO: This and the normal sync share a ton of code. It's hard to really... figure out the right way to 
     private Pair<Integer, String> recover(HttpRequestEndpoints requestor, AndroidTransactionParserFactory factory) {
-        this.publishProgress(PROGRESS_RECOVERY_NEEDED);
-
-        Logger.log(AndroidLogger.TYPE_USER, "Sync Recovery Triggered");
+        while (asyncRestoreHelper.retryWaitPeriodInProgress()) {
```
Comment (ctsims) [inline app/src/org/commcare/tasks/DataPullTask.java]: There should be a sleep() after this call so this thread doesn't stay redlined while we wait.

File: app/unit-tests/src/org/commcare/android/tests/DataPullTaskTest.java (hunk 1 above)
Comment (amstone326) [inline app/unit-tests/src/org/commcare/android/tests/DataPullTaskTest.java]: Ha yea I was so pleased with how easy it was to write this test

File: app/src/org/commcare/tasks/DataPullTask.java (hunk 2 above)
Comment (amstone326) [inline app/src/org/commcare/tasks/DataPullTask.java]: For how long?

Comment (ctsims) [inline app/src/org/commcare/tasks/DataPullTask.java]: It will be full seconds (many of them) for the retry period, so probably 500ms or so?

-Clayton

Comment (amstone326) [inline app/src/org/commcare/tasks/DataPullTask.java]: Updated, and I also added a sleep to the parallel while loop in `getRequestResultOrRetry()`

**Pull request merged**